proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -869,6 +869,194 @@ ref struct S<T>
+
+Note that this is not meant to be a 100% complete documentation. Documenting every single behavior isn't a goal here. Instead it's meant to establish a general understanding and common verbiage by which the model, and potential changes to it, can be discussed.
+
+Usually it's not necessary to directly talk about lifetime types. The exceptions are places where lifetimes can vary based on particular "instantiation" sites. This is a kind of polymorphism and we call these varying lifetimes "generic lifetimes", represented as generic parameters. C# does not provide syntax for expressing lifetime generics, so we define an implicit "translation" from C# to an expanded lowered language that contains explicit generic parameters.
+The below examples make use of named lifetimes. The syntax `$a` refers to a lifetime named `a`. It is a lifetime that has no meaning by itself but can be given a relationship to other lifetimes via the `where lifetime` syntax. There are a few predefined lifetimes for convenience and brevity below:
+
+- `$heap`: this is the lifetime of any value that exists on the heap. It is available in all scopes and method signatures.
+- `$local`: this is the lifetime of any value that exists on the method stack. It's effectively a name place holder for *current method*. It is implicitly defined in methods and can appear in method signatures except for any output position.
+- `$ro`: name place holder for the *return only* safe to escape scope
+- `$cm`: name place holder for the *calling method* safe to escape scope
+
+There are a few predefined relationships between lifetimes:
+
+- `lifetime $heap >= $a` for all lifetimes `$a`
+- `lifetime $cm >= $ro` 
+- `lifetime $x >= $local` for all predefined lifetimes. User defined lifetimes have no relationship to local unless explictly defined.
+
+Lifetime variables when defined on types can be invariant or covariant. These are expressed using the same syntax as generic parameters:
+
+```csharp
+// $this is covariant
+// $a is invariant
+ref struct S<out $this, out $a> 
+```
+
+The lifetime of a ref is expressed by providing a lifetime argument to the ref. For example a `ref` that refers to the heap is expressed as `ref<$heap>`.
+
+When defining a constructor in the model the name `new` will be used for the method. It is necessary to have a parameter list for the returned value as well as the constructor arguments. This is necessary to express the relationship between constructor inputs and the constructed value. Rather than having `Span<$a><$ro>` the model will use `Span<$a> new<$ro>` instead. The type of `this` in the constructor, including lifetimes, will be the defined return value.
```

Comment:
Did you consider representing `this` in a constructor as an `out` parameter here, as it's closer to how it's represented in the symbol model.

Reply:
I thought about it a bit but wasn't sure if that was a step too far. It's somewhat natural for `struct` types but not as much for `class` types. I could be convinced to change this though.